File: Intermed/functionRef.cpp
#include <iostream>

// Function to swap two integers using pass by reference
void swap(int& a, int& b) {
    int temp = a;
    a = b;
    b = temp;
}

int main() {
    int x = 5;
    int y = 10;

    std::cout << "Before swapping: x = " << x << ", y = " << y << std::endl;

    // Call the swap function, passing x and y by reference
    swap(x, y);

    std::cout << "After swapping: x = " << x << ", y = " << y << std::endl;

    return 0;
}

//print 5, 10
//print 10, 5

File: Intermed/functionRef.py
def swap(a, b):
    # Python swaps can be done directly without needing a temporary variable
    a, b = b, a
    return a, b

def main():
    x = 5
    y = 10

    print("Before swapping: x =", x, ", y =", y)

    # Call the swap function, passing x and y by reference
    x, y = swap(x, y)

    print("After swapping: x =", x, ", y =", y)

if __name__ == "__main__":
    main()

